oak_crypto/src/verifier.rs

```diff
@@ -0,0 +1,32 @@
+//
+// Copyright 2023 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use anyhow::Context;
+
+pub trait Verifier {
+    fn verify(&self, message: &[u8], signature: &crate::signer::Signature) -> anyhow::Result<()>;
+}
+
+impl Verifier for p256::ecdsa::VerifyingKey {
+    fn verify(&self, message: &[u8], signature: &crate::signer::Signature) -> anyhow::Result<()> {
+        let parsed_signature = p256::ecdsa::Signature::from_slice(&signature.signature)
+            .map_err(anyhow::Error::msg)
+            .context("could not parse signature")?;
+        <p256::ecdsa::VerifyingKey as p256::ecdsa::signature::Verifier<
+            p256::ecdsa::Signature,
+        >>::verify(&self, message, &parsed_signature).map_err(anyhow::Error::msg)
```

Comment:
![this is the correct rust syntax. you may not like it but this is what peak performance looks like](https://github.com/project-oak/oak/assets/8875406/f0606bfc-d6e2-467b-b644-754296c66d6a)

this meme, whose contents may or may not infringe upon the copyright of Marvel Studios, is brought to you by artificial intelligence.

Reply:
took me a hot second to figure out the right syntax here. We have to invoke this trait explicitly since a trait with the same method is already in scope.

Reply:
is that... iron man made of clay?

Reply:
> is that... iron man made of clay?

i think it's supposed to represent a rusty iron man? the specific prompt was "background for a meme about the rust programming language". All four images generated featured some version of iron man smh. ^^

Reply:
/me nods slowly